metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricQueryBuilder.java

```diff
@@ -0,0 +1,267 @@
+            LiteralConstant.newBuilder()
+                .setValue(Value.newBuilder().setString(rhsExpression.getStringValue()))
+                .build());
+      default:
+        throw new IllegalArgumentException("error");
+    }
+  }
+
+  static Operator convertFilterValueOperator(ValueOperator valueOperator) {
+    switch (valueOperator) {
+      case VALUE_OPERATOR_EQ:
+        return Operator.EQ;
+      default:
+        throw new IllegalArgumentException("error");
+    }
+  }
+
+  static Operator convertFilterLogicalOperator(LogicalOperator logicalOperator) {
+    switch (logicalOperator) {
+      case LOGICAL_OPERATOR_OR:
+        return Operator.OR;
+      case LOGICAL_OPERATOR_AND:
+        return Operator.AND;
+      default:
+        throw new IllegalArgumentException("error");
+    }
+  }
+
+  static org.hypertrace.core.query.service.api.Filter addTimeFilter(
+      org.hypertrace.core.query.service.api.Filter existingFilter,
+      long startTime,
+      long endTime,
+      String timeStampAttributeId) {
+    org.hypertrace.core.query.service.api.Filter.Builder timeFilterBuilder =
+        org.hypertrace.core.query.service.api.Filter.newBuilder()
+            .setOperator(Operator.AND)
+            .addChildFilter(createLongFilter(timeStampAttributeId, Operator.GE, startTime))
+            .addChildFilter(createLongFilter(timeStampAttributeId, Operator.LT, endTime));
+
+    if (null == existingFilter) {
+      return timeFilterBuilder.build();
+    }
+
+    return org.hypertrace.core.query.service.api.Filter.newBuilder()
+        .setOperator(Operator.AND)
+        .addChildFilter(timeFilterBuilder.build())
+        .addChildFilter(existingFilter)
+        .build();
+  }
+
+  public static org.hypertrace.core.query.service.api.Filter createLongFilter(
+      String columnName, Operator op, long value) {
+    return org.hypertrace.core.query.service.api.Filter.newBuilder()
+        .setLhs(createColumnExpression(columnName))
+        .setOperator(op)
+        .setRhs(createLongLiteralExpression(value))
+        .build();
+  }
+
+  public static Expression createLongLiteralExpression(long value) {
+    return Expression.newBuilder()
+        .setLiteral(
+            LiteralConstant.newBuilder()
+                .setValue(Value.newBuilder().setValueType(ValueType.LONG).setLong(value)))
+        .build();
+  }
+
+  static void setSelection(
+      QueryRequest.Builder queryBuilder,
+      Attribute metricAttribute,
+      MetricAggregationFunction metricAggregationFunction) {
+    Function function =
+        Function.newBuilder()
+            .setFunctionName(convertToQueryRequestFunctionName(metricAggregationFunction))
+            .addArguments(
+                Expression.newBuilder()
+                    .setColumnIdentifier(
+                        ColumnIdentifier.newBuilder()
+                            .setColumnName(
+                                metricAttribute.getScope() + "." + metricAttribute.getKey())
```

Comment:
can we use the string joiner function here?

Reply:
used